test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -601,30 +595,38 @@ class test_ucp_tag_nbx : public test_ucp_tag {
                                   const ucp_request_param_t &sparam,
                                   const ucp_request_param_t &rparam)
     {
-        int rcache_count;
-
         if (sparam.op_attr_mask & UCP_OP_ATTR_FIELD_REQUEST) {
             /* Can't send multiple requests for a single user request memory
                handle */
             return;
         }
 
+        int rcache_prev;
+        rcache_prev = UCS_STATS_GET_COUNTER(sender().ucph()->rcache->stats,
+                                            UCS_RCACHE_GETS);
+
         for (int i = 0; i < 10; ++i) {
             do_send_recv(send_dt, recv_dt, sparam, rparam);
         }
 
-        rcache_count = UCS_STATS_GET_COUNTER(sender().ucph()->rcache->stats,
-                                             UCS_RCACHE_GETS);
+        int rcache_curr;
+        rcache_curr = UCS_STATS_GET_COUNTER(sender().ucph()->rcache->stats,
+                                            UCS_RCACHE_GETS);
 
-        /* mem_map uses rcache twice ('lookup + get') */
-        EXPECT_EQ(2, rcache_count);
+        /* Compare counters before and after iterations */
+        EXPECT_EQ(rcache_prev, rcache_curr);
     }
 
     void test_recv_send(size_t size, const void *send_buffer, void *recv_buffer,
                         ucp_request_param_t send_param = null_param,
                         ucp_request_param_t recv_param = null_param,
                         completion_value completion = null_completion)
     {
+        if (prereg() && sender().ucph()->config.ext.proto_request_reset) {
+            UCS_TEST_SKIP_R(
+                    "prereg is not supported with 'proto_request_reset'");
```

Comment:
why not supported?

Reply:
AFAIR proto_request_reset resets user memh

Reply:
can we fix it too, to retain user memh?